src/tests/trainers.test.js

```diff
@@ -1,3 +1,274 @@
+    expect(response.status).toBe(200);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.data).toBeDefined();
+    expect(response.body.error).toBeFalsy();
+  });
+});
+
+describe('POST /api/trainers', () => {
+  test('Should create a trainer', async () => {
+    const response = await request(app).post('/api/trainers').send(newTrainer);
+    expect(response.status).toBe(201);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeFalsy();
+  });
+
+  test('Should return The admin with DNI already exists', async () => {
+    const reqBody = {
+      dni: 43576575,
+    };
+    const response = await request(app).post('/api/trainers').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Should return The admin with this Email already exists', async () => {
+    const reqBody = {
+      email: 'delfi@gmail.com',
+    };
+    const response = await request(app).post('/api/trainers').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Should return invalid name format', async () => {
+    const reqBody = {
+      firstName: '123Gino',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return invalid phone number fortmat', async () => {
+    const reqBody = {
+      phone: '8383',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return invalid Password format', async () => {
+    const reqBody = {
+      password: 'strongepassword123',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return invalid salary format', async () => {
+    const reqBody = {
+      salary: 'salary',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return an status 400', async () => {
+    const reqBody = {};
+    const response = await request(app).post('/api/trainers').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+});
+
+describe('DELETE /api/trainers', () => {
+  test('Should create a trainer with a status 200', async () => {
```

Comment:
I think you meant to put "should delete"

Reply:
@Ginoboca1